module/zfs/zfs_quota.c

```diff
@@ -286,6 +286,7 @@ zfs_set_userquota(zfsvfs_t *zfsvfs, zfs_userquota_prop_t type,
 	int err;
 	dmu_tx_t *tx;
 	uint64_t *objp;
+	uint64_t zap_nentries, prev_objp;
```

Comment:
nit: I'd move the declaration for `zap_nentries` in to the `if (quota == 0) {` block to reduce its scope.